src/Engine/UnitActionType.java

```diff
@@ -303,6 +307,55 @@ private Object readResolve()
     }
   }
 
+  public static class RepairUnit extends UnitActionType
+  {
+    @Override
+    public GameActionSet getPossibleActions(GameMap map, Path movePath, Unit actor, boolean ignoreResident)
+    {
+      XYCoord moveLocation = new XYCoord(movePath.getEnd().x, movePath.getEnd().y);
+      if( ignoreResident || map.isLocationEmpty(actor, moveLocation) )
+      {
+        ArrayList<GameAction> repairOptions = new ArrayList<GameAction>();
+        ArrayList<XYCoord> locations = Utils.findLocationsInRange(map, moveLocation, 1, 1);
+
+        // For each location, see if there is a friendly unit to repair.
+        for( XYCoord loc : locations )
+        {
+          // If there's a friendly unit there who isn't us, we can repair them.
+          Unit other = map.getLocation(loc).getResident();
+          if( other != null && other.CO == actor.CO && other != actor &&
```

Comment:
Should we allow repairing allies?

Reply:
I think *probably* what we should do is allow allies to repair themselves on your property. I think that's how DoR does it (might be wrong though)

Reply:
Yes. In DoR, you repair units on allies' properties at the start of your turn, just like your own.

Not sure why we wouldn't allow allies to heal each other too (though in that case, the one doing the action should pay, unlike the passive turn-init healing), though I'm not sure if AWDS allows that.